vtr_flow/scripts/python_libs/vtr/inspect.py

```diff
@@ -11,6 +14,465 @@
+            ) from ValueError
+
+        # Get relative ratio
+        norm_check_value = None
+        if golden_value == 0.0:  # Avoid division by zero
+            if golden_value == check_value:
+                return True, "{} and check both equal 0".format(check_string)
+            norm_check_value = float("inf")
+        else:
+            norm_check_value = check_value / golden_value
+
+        # Check if the original values match
+        if original_check_value == original_golden_value:
+            return True, "Check value equal to {}".format(check_string)
+
+        # Check if value within range
+        if (self.min_value() <= norm_check_value <= self.max_value()) or abs(
+            check_value - golden_value
+        ) <= self.abs_threshold():
+            return True, "relative value within range"
+
+        return (
+            False,
+            "relative value {} outside of range [{},{}], "
+            "above absolute threshold {} and not equal to {} value: {}".format(
+                norm_check_value,
+                self.min_value(),
+                self.max_value(),
+                self.abs_threshold(),
+                check_string,
+                golden_value,
+            ),
+        )
+
+
+class ParseResults:
+    """ This class contains all parse results and metrics """
+
+    PRIMARY_KEYS = ("architecture", "circuit", "script_params")
+
+    def __init__(self):
+        self._metrics = OrderedDict()
+
+    def add_result(self, arch, circuit, parse_result, script_param=None):
+        """ Add new parse result for given arch/circuit pair """
+        script_param = load_script_param(script_param)
+        self._metrics[(arch, circuit, script_param)] = parse_result
+
+    def metrics(self, arch, circuit, script_param=None):
+        """ Return individual metric based on the architechure, circuit and script"""
+        script_param = load_script_param(script_param)
+        if (arch, circuit, script_param) in self._metrics:
+            return self._metrics[(arch, circuit, script_param)]
+        return None
+
+    def all_metrics(self):
+        """ Return all metric results """
+        return self._metrics
+
+
+def load_script_param(script_param):
+    """
+    Create script parameter string to be used in task names and output.
+    """
+    if script_param and "common" not in script_param:
+        script_param = "common_" + script_param
+    if script_param:
+        script_param = script_param.replace(" ", "_")
+    else:
+        script_param = "common"
+    return script_param
+
+
+def load_parse_patterns(parse_config_filepath):
+    """
+    Loads the parse patterns from the desired file.
+    These parse patterns are later used to load in the results file
+    """
+    parse_patterns = OrderedDict()
+
+    for line in load_config_lines(parse_config_filepath):
+
+        components = line.split(";")
+
+        if len(components) == 3 or len(components) == 4:
```

Comment:
An example of the input format would be useful here as documentation e.g.

```
name path regex [default value]
```

Reply:
This has been fixed